Kuadrant/kuadrant-controller
# utilise wasm as the integration point

This PR replaces Envoy's ratelimit filter with [wasm-shim](https://github.com/Kuadrant/wasm-shim/) and few changes in the RateLimitPolicy that are needed and allowed by this change.

## Verification Steps

1. `make local-setup`
2. `kubectl apply -f examples/toystore/toystore.yaml`
3. `kubectl apply -f examples/toystore/virtualService.yaml`
4. `kubectl apply -f config/samples/apim_v1alpha1_ratelimitpolicy.yaml`
5. `kubectl annotate virtualservice/toystore kuadrant.io/ratelimitpolicy=toystore`
6. `k port-forward -n kuadrant-system svc/kuadrant-gateway 9080:80`
7. `curl -v -H 'Host: api.toystore.com' http://localhost:9080/toy`

You can also check config dump with pre/post auth wasm filters with different configs.
